packages/happtiq_commons_google_cloud/src/happtiq_commons_google_cloud/gcs_api_service.py

```diff
@@ -37,3 +37,13 @@ def upload_file(self, file_path: bytes,  bucket_name: str, destination: str, con
         blob.upload_from_filename(file_path, content_type=content_type, timeout=self.timeout)
         self.logger.info(f"File uploaded to {destination}.")
         return f"gs://{bucket_name}/{destination}"
+    
+    def list_files_with_prefix(self, bucket_name: str, prefix: str):
```

Comment:
missing unit tests ;)

Reply:
unit testing added, please review it. I really don't know about the issue that I have all previous changes also having on this branch. Maybe I created branch based on other branch instead of main. I have no idea how it happens.. I dont want to adjust to much so I dont have any change get lost :) can we keep on working like this this time?

Reply:
no. you should learn to create branches and commit correctly. which means fixing the commits and branches if they are not correct

Reply:
git rebase -i main  --> opens the editor(wim) for interactive rebasing - so we can pick/drop the commits from the past. and then :wq to save the changes. (for me a recap)